src/noisepy/seis/zarrstore.py

```diff
@@ -0,0 +1,154 @@
+        contains = self._contains(timespan, station_pair, channel_pair)
+        if contains:
+            logger.info(f"Cross-correlation {station_pair} and {channel_pair} already exists")
+        return contains
+
+    def save_parameters(self, parameters: ConfigParameters):
+        fc_metadata = os.path.join(self.directory, "fft_cc_data.json")
+
+        fout = open(fc_metadata, "w")
+        # WIP actually serialize this
+        fout.write(str(parameters.__dict__))
+        fout.close()
+
+    def append(
+        self,
+        timespan: DateTimeRange,
+        src_chan: Channel,
+        rec_chan: Channel,
+        params: ConfigParameters,
+        cc_params: Dict[str, Any],
+        corr: np.ndarray,
+    ):
+        # source-receiver pair: e.g. CI.ARV_CI.BAK
+        station_pair = self._get_station_pair(src_chan, rec_chan)
+        # channels, e.g. bhn_bhn
+        channels = f"{src_chan.type.name}_{rec_chan.type.name}"
+        data = np.zeros(corr.shape, dtype=corr.dtype)
+        data[:] = corr[:]
+        self._add_aux_data(timespan, cc_params, station_pair, channels, data)
+
+    def mark_done(self, timespan: DateTimeRange):
+        self._add_aux_data(timespan, {}, PROGRESS_DATATYPE, DONE_PATH, np.zeros(0))
+
+    def is_done(self, timespan: DateTimeRange):
+        done = self._contains(timespan, PROGRESS_DATATYPE, DONE_PATH)
+        if done:
+            logger.info(f"Timespan {timespan} already computed")
+        return done
+
+    def read(self, chan1: Channel, chan2: Channel, start: datetime, end: datetime) -> np.ndarray:
+        pass
+
+    # private helper methods
+    def _contains(self, timespan: DateTimeRange, data_type: str, path: str = None):
+        filename = self._get_filename(timespan)
+        if not os.path.isfile(filename):
+            return False
+
+        store = zarr.DirectoryStore(filename)
+        with zarr.open(store, mode="r") as ccf_ds:
+            # source-receiver pair
+            exists = data_type in ccf_ds.attrs["auxiliary_data"]
+            if path is not None and exists:
+                return path in ccf_ds.attrs["auxiliary_data"][data_type]
+            return exists
+
+    def _add_aux_data(self, timespan: DateTimeRange, params: Dict, data_type: str, path: str, data: np.ndarray):
+        filename = self._get_filename(timespan)
+        store = zarr.DirectoryStore(filename)
+        with zarr.open(store, mode="a") as ccf_ds:
+            ccf_ds.attrs.setdefault("auxiliary_data", {}).setdefault(data_type, {})
+            ccf_ds.attrs["auxiliary_data"][data_type][path] = {"parameters": params}
+            ccf_ds.create_dataset(f"auxiliary_data/{data_type}/{path}", data=data)
+
+    def _get_station_pair(self, src_chan: Channel, rec_chan: Channel) -> str:
+        return f"{src_chan.station}_{rec_chan.station}"
+
+    def _get_channel_pair(self, src_chan: Channel, rec_chan: Channel) -> str:
+        return f"{src_chan.type.name}_{rec_chan.type.name}"
```

Comment:
Can you move these into `stores.py` and share them between `ASDFCCStore` and `ZarrCCStore` to avoid duplication? Note that these now take `Station` and `ChannelType` args respectively.